### PR TITLE
Refactor pbp

### PR DESCRIPTION
**Pull request by patstrom**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -161,6 +161,46 @@ The SDK will try to perform as many requests as possible concurrently. It will c
...
+
+```go
+// Note that for this you have to import the structs sub-package
+import . "github.com/AbiosGaming/go-sdk-v2/structs"
+
+team, _ := a.MatchesById(301281, nil)
+switch pbp := team.TeamStats.PlayByPlay.(type) {
+	case DotaTeamStats:
+		// Do something
+		break
```
Comment (johlo) [inline README.md]: No need for the `break`.  It's nice to make it explicit in the example but not really go'ish.

File: structs/match_summary.go [hunk 2]
```diff
@@ -121,13 +118,46 @@ type DotaMatchSummaryStruct struct {
...
+				Slot3 DotaItemStruct `json:"slot_3"`
+				Slot4 DotaItemStruct `json:"slot_4"`
+				Slot5 DotaItemStruct `json:"slot_5"`
+				Slot6 DotaItemStruct `json:"slot_6"`
+			} `json:"inventory"`
+			Backpack struct {
+				Slot1 DotaItemStruct `json:"slot_1"`
+				Slot2 DotaItemStruct `json:"slot_2"`
+				Slot3 DotaItemStruct `json:"slot_3"`
+			} `json:"backback"`
```
Comment (johlo) [inline structs/match_summary.go]: Should be `json:"backpack"`

File: structs/match_summary.go [hunk 3]
```diff
@@ -143,3 +173,219 @@ type DotaDmg struct {
...
+		ToHeroes     LolDmgStruct `json:"to_heroes"`
+		DamageTaken  LolDmgStruct `json:"damage_taken"`
+		LargestCrit  int64        `json:"largest_crit"`
+		ToObjectives int64        `json:"to_objectives"`
+		ToTurrets    int64        `json:"to_turrets"`
+	} `json:"damage"`
+	Support struct {
+		AmountHealed     int64 `json:"amount_healed"`
+		UnitsHealed      int64 `json:"units_healed"`
+		CrowsControlTime int64 `json:"crowd_control_time"`
```
Comment (johlo) [inline structs/match_summary.go]: `CrowsControlTime` -> `CrowdControlTime`

File: structs/series.go [hunk 4]
```diff
@@ -32,4 +34,59 @@ type SeriesStruct struct {
...
+	delete(partial, "summary")
+	data, _ = json.Marshal(partial)
+
+	var ss seriesStruct
+	if err := json.Unmarshal(data, &ss); err != nil {
+		return err
+	}
+
+	// "null" is 4 bytes
+	if len(summary) > 4 {
```
Comment (johlo) [inline structs/series.go:62]: Maybe explicitly check for "null" instead?

**Review by johlo — CHANGES_REQUESTED** [review]
Approach looks good!

File: structs/series.go (hunk 4 above)
Comment (patstrom) [reply structs/series.go:62]: There can be no useful data that has `len <= 4` anyways so my thought process was that it doesn't matter if it's explicitly set to "null". This way there is no need for a string comparison (although the performance difference is most likely negligible).

**Review by johlo — APPROVED** [review]
❤️💚💙💛💜🦄 👑 🦄 💜💛💙💚❤️

**Pull request merged**